### ISSUE TITLE
Profiling server starts when running migrations, console, etc.

### ISSUE THREAD

**Issue by gmcgibbon** [Shopify/app_profiler#66]
It should only start on server boot / `bin/rails server`

https://github.com/Shopify/app_profiler/pull/48#discussion_r984068012

**Comment by dalehamel** [comment]
One solution on the app side is to only enable it for production (or staging) with something like:

```
unless Rails.env.development? || Rails.env.test?
  config.app_profiler.server_enabled = true
end
```

**Comment by dalehamel** [comment]
I think that things are basically working as intended - it is up to the app to decide what environments / situations they want the server enabled, it defaults to disabled. On our end, we are only enabling it with `Rails.env.production?` in many cases at the moment.

We can leave this open for visibility, or close it. WDYT @gmcgibbon

**Comment by gmcgibbon** [comment]
We run production consoles sometimes, so I think we could be a little more clever about it. We can use `if defined?(Rails::Server)`, but that excludes job workers. Job running depends on what adapter Active Job is using so there isn't a great way to determine if a process was started for running jobs without checking for adapter specific constants.

Since this is disabled by default I think we can close this for now. If someone runs into this issue, they can start the server manually.

**Comment by dalehamel** [comment]
> We run production consoles sometimes, so I think we could be a little more clever about it.

Yeah i had considered this case too, but I think that it is fine as the profile server will just log a line saying it has started, which shouldn't impact the rails console usage otherwise. It may add some small number of microseconds to the startup time as it needs to do the syscall to bind to the unix socket, but aside from the log message I don't see this as being much overhead, especially given that the rails console is invoked by a human in the exceptional case, and not any machine process. I think that this issue was particularly prevalent and noticeable in development, but only when the server is started unconditional in development.
